src/krux/psbt.py

```diff
@@ -504,35 +519,43 @@ def _fill_zero_fingerprint_scope(self, scope):
                     filled += 1
         return filled
 
-    def sign(self):
-        """Signs the PSBT removing all irrelevant data"""
+    def sign(self, trim=True):
+        """Signs the PSBT and preserves necessary fields for the final transaction"""
         self.add_signatures()
 
+        if not trim:
+            return
+
         trimmed_psbt = PSBT(self.psbt.tx)
         for i, inp in enumerate(self.psbt.inputs):
-            # Copy the final_scriptwitness if it's present (Taproot case)
+            # Copy the final_scriptwitness if present
             if inp.final_scriptwitness:
                 trimmed_psbt.inputs[i].final_scriptwitness = inp.final_scriptwitness
-            # Copy partial signatures for multisig or other script types
+
+            # Copy any partial signatures
             if inp.partial_sigs:
                 trimmed_psbt.inputs[i].partial_sigs = inp.partial_sigs
 
-            # Include the PSBT_IN_WITNESS_UTXO field if it exists
-            if hasattr(inp, "witness_utxo"):
+            # Preserve witness UTXO if present
+            if inp.witness_utxo:
                 trimmed_psbt.inputs[i].witness_utxo = inp.witness_utxo
 
-            # Include the PSBT_IN_NON_WITNESS_UTXO field if it exists (Legacy)
-            if hasattr(inp, "non_witness_utxo"):
+            # Preserve non-witness UTXO if present (for legacy inputs)
+            if inp.non_witness_utxo:
                 trimmed_psbt.inputs[i].non_witness_utxo = inp.non_witness_utxo
 
-            # Check for P2SH (Nested SegWit) and copy redeem_script if present
-            if hasattr(inp, "redeem_script"):
+            # Preserve redeem_script for P2SH or nested SegWit
+            if inp.redeem_script:
                 trimmed_psbt.inputs[i].redeem_script = inp.redeem_script
 
-            # Check for P2WSH (SegWit multisig) and copy witness_script if present
-            if hasattr(inp, "witness_script"):
+            # Preserve witness_script for P2WSH multisig
+            if inp.witness_script:
                 trimmed_psbt.inputs[i].witness_script = inp.witness_script
 
+            # Preserve taproot_key_sig for Taproot inputs
+            if inp.taproot_key_sig:
+                trimmed_psbt.inputs[i].taproot_key_sig = inp.taproot_key_sig
+
```

Comment:
`inp.taproot_script_path_sigs` too?

I've seen this field (and others) in the "sign to sdcard" version but it's not showing up in the "sign to qr" version.

Reply:
Indeed, `inp.taproot_sigs` were missing for QR code exported signed PSBT. Fixed!
Now we probably still need to add some more fields so the signed PSBT can be broadcasted as a standalone verifiable and broadcastable PSBT (not only for  updating a full PSBT on the coordinator)